src/libraries/Microsoft.Extensions.Logging/src/LoggerFactoryScopeProvider.cs

```diff
@@ -48,12 +47,48 @@ void Report(Scope current)
                     }
 
                     callback(activityLogScope, state);
+
+                    // Tags and baggage are opt-in and thus we assume that most of the time it will not be used.
+                    if ((_activityTrackingOption & ActivityTrackingOptions.Tags) != 0)
+                    {
+                        const string additionalItemsTagsPropertyKey = "__ActivityTagItemsLogScope__";
+                        AdditionalActivityItemsLogScope<object?> scope = LogScopeForAdditionalActivityItems(activity, activity.TagObjects, additionalItemsTagsPropertyKey);
+                        // Only add the scope if it's not empty
+                        if (scope.GetEnumerator().MoveNext())
```

Comment:
This feels a bit ugly. However, if we don't to this new scopes empty scopes are added if baggage and tags are empty but logging is turned on.
A second option would be to change the logger formatters to not add string.Empty scopes

https://github.com/dotnet/runtime/blob/79ae74f5ca5c8a6fe3a48935e85bd7374959c570/src/libraries/Microsoft.Extensions.Logging.Console/src/JsonConsoleFormatter.cs#L115
https://github.com/dotnet/runtime/blob/79ae74f5ca5c8a6fe3a48935e85bd7374959c570/src/libraries/Microsoft.Extensions.Logging.Console/src/SimpleConsoleFormatter.cs#L195

Reply:
@tarekgh plz let me know your opinion on this.